CompuCell3D/core/CompuCell3D/Field3D/Point3D.h

```diff
@@ -44,66 +42,47 @@ namespace CompuCell3D {
     /** 
      * Construct a point at the origin.
      */    
-    Point3D() : x(0), y(0), z(0) {}
+    Point3D();
 
-    Point3D(const short x, const short y, const short z) :
-      x(x), y(y), z(z) {}
+    Point3D(const short x, const short y, const short z);
 
     /** 
      * Copy constructor
      */
-    Point3D(const Point3D &pt) : x(pt.x), y(pt.y), z(pt.z) {}
+    Point3D(const Point3D &pt);
 
     /** 
      * Assignment operator.
      */    
-    Point3D &operator=(const Point3D pt) {
-      x = pt.x;
-      y = pt.y;
-      z = pt.z;
-      return *this;
-    }
+    Point3D &operator=(const Point3D pt);
 
     /** 
      * Add the coordinates of pt to this Point3D.
      */
-    Point3D &operator+=(const Point3D pt) {
-      x += pt.x;
-      y += pt.y;
-      z += pt.z;
-      return *this;
-    }
+    Point3D &operator+=(const Point3D pt);
 
     /** 
      * Subtract the coordinates of pt to this Point3D.
      */
-    Point3D &operator-=(const Point3D pt) {
-      x -= pt.x;
-      y -= pt.y;
-      z -= pt.z;
-      return *this;
-    }
+    Point3D &operator-=(const Point3D pt);
         
     /// Comparison operator
-    bool operator==(const Point3D pt) const {
-      return (x == pt.x && y == pt.y && z == pt.z);
-    }
+    bool operator==(const Point3D pt) const;
+
     /// Not equal operator
-    bool operator!=(const Point3D pt) const {
-		
-      return !(*this==pt);
-    }
+    bool operator!=(const Point3D pt) const;
     
-   bool operator<(const Point3D  _rhs) const{
-      return x < _rhs.x || (!(_rhs.x < x)&& y < _rhs.y)
-			||(!(_rhs.x < x)&& !(_rhs.y <y )&& z < _rhs.z);
-   }
-    //// Begin XMLSerializable interface
-    //virtual void readXML(XMLPullParser &in);
-    //virtual void writeXML(XMLSerializer &out);
-    //// End XMLSerializable interface
+    bool operator<(const Point3D _rhs) const;
     
     friend std::ostream &operator<<(std::ostream &stream, const Point3D &pt);
+
+    // Python support
+
+#ifdef SWIGPYTHON
```

Comment:
I wonder if perhaps a better approach would be to define this function in the SWIG CompuCell.i interface file - I guess you would use %{extend } directive, something like that is used for CellG class. If this is not too much trouble let's try this approach. I think what I am driving at is that I would like to keep CC3D API separate from depending on Python too much. (yes I know we already have few references to Python in e.g. CellG but I think since we are using SWIG we may want to take advantage of the extend functionality that it offers). What do you think?

Reply:
I prefer class definitions to be localized as much as possible, rather than maintaining a giant monolithic script like CompuCell.i. This could be implemented in a number of ways, however, whether like adding python dictionaries with preprocessor directives as in CellG.h, [FocalPointPlasticity.h](https://github.com/CompuCell3D/CompuCell3D/blob/develop/CompuCell3D/core/CompuCell3D/plugins/FocalPointPlasticity/FocalPointPlasticityLinks.h) and done here, or expanding our usage of swig files as in [DerivedProperty.i](https://github.com/CompuCell3D/CompuCell3D/blob/master/CompuCell3D/core/pyinterface/CompuCellPython/DerivedProperty.i). For now, I can just move the implementation to CompuCell.i, as suggested.